design/src/rfcs/rfc0016_body_callback_apis_v2.md

```diff
@@ -74,21 +77,26 @@ impl Crc32Callback {
 
     fn trailers(&self) -> Result<Option<HeaderMap<HeaderValue>>, BoxError> {
         let mut header_map = HeaderMap::new();
-        let key = HeaderName::from_static(CRC_32_HEADER_NAME);
-        // We clone the hasher because `Hasher::finalize` consumes `self`
-        let hash = self.hasher.clone().finalize();
-        let value = HeaderValue::from_str(&base64::encode(u32::to_be_bytes(hash)))
-            .expect("base64 will always produce valid header values from checksums");
-
-        header_map.insert(key, value);
+        header_map.insert(Self::header_name(), self.header_value());
 
         Ok(Some(header_map))
     }
 
     // Size of the checksum in bytes
-    fn size() -> u64 {
+    fn size() -> usize {
         4
```

Comment:
I think this number is not actually accurate (should be 8?) since you need the length of the base64 encoding of the final number. Do we know for certain that that encoded version of it will always be the same length? I _think_ it is due to the padding, but I'm not familiar enough off the top of my head to say with certainty.